src/lib/functions/runtimes/index.ts

```diff
@@ -1,30 +1,70 @@
+import type { ExtendedRoute, Route } from '@netlify/zip-it-and-ship-it'
+
+import type { BuildCommandCache } from '../memoized-build.js'
+import type NetlifyFunction from '../netlify-function.js'
+import type { NormalizedCachedConfigConfig } from '../../../utils/command-helpers.js'
+
 import * as go from './go/index.js'
+import type { GoInvokeFunctionResult } from './go/index.js'
 import * as js from './js/index.js'
+import type { JsInvokeFunctionResult } from './js/index.js'
 import * as rust from './rust/index.js'
+import type { RustInvokeFunctionResult } from './rust/index.js'
+
+export type BaseBuildResult = {
+  includedFiles?: undefined | string[]
+  mainFile?: undefined | string
+  outputModuleFormat?: undefined | string
+  schedule?: undefined | string
+  srcFiles: string[]
+
+  // TODO(serhalp) This module and type shouldn't know about these zisi types. Refactor to allow the JS runtime's zisi
+  // builder to define this on its extended base build result type.
```

Comment:
The way this runtime stuff is typed could really be improved, but I had to just cut my losses here and ship something that is strongly typed and works. We can always follow up.